### PR TITLE
Pd 116/tds state definitions

### PR DESCRIPTION
**Pull request by LeandroParisi**
1. Implements TDS state management
2. Implements GRPC procedures to manage and get TDS state

### CODE REVIEW

File: src/ProtobufInterfaces/proto/tds.proto [hunk 1]
```diff
@@ -8,12 +8,214 @@ package ixl;
...
+  MsgAdditionalInformation msgAdditionalInformation = 17;
+  PomStatus pomStatus = 18;
+  PowerSupply powerSupply = 19;
+  State state = 20;
+  StateOfPassing stateOfPassing = 21;
+  TvpsOccupancyStatus tvpsOccupancyStatus = 22;
+  TvpsOccupancyStatusBefore tvpsOccupancyStatusBefore = 23;
+  TvpsOccupiedByAtLeastOneWheel tvpsOccupiedByAtLeastOneWheel = 24;
+  TvpsReasonForDisturbance tvpsReasonForDisturbance = 25;
+  PdiConection pdiConection = 26;
```
Comment (rs22) [inline src/ProtobufInterfaces/proto/tds.proto]: there's a typo here (should be `PdiConnection`). I know that this is probably generated/derived from test data, but we should fix this in the test as well

File: src/TrainDetectionSystem/Components/TDS/TDSState.cs [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        public TDSStateMessage.MsgAdditionalInformation MsgAdditionalInformation { get; set; }
+        public TDSStateMessage.PomStatus PomStatus { get; set; }
+        public TDSStateMessage.PowerSupply PowerSupply { get; set; }
+        public TDSStateMessage.State State { get; set; }
+        public TDSStateMessage.StateOfPassing StateOfPassing { get; set; }
+        public TDSStateMessage.TvpsOccupancyStatus TvpsOccupancyStatus { get; set; }
+        public TDSStateMessage.TvpsOccupancyStatusBefore TvpsOccupancyStatusBefore { get; set; }
+        public TDSStateMessage.TvpsOccupiedByAtLeastOneWheel TvpsOccupiedByAtLeastOneWheel { get; set; }
+        public TDSStateMessage.TvpsReasonForDisturbance TvpsReasonForDisturbance { get; set; }
+        public TDSStateMessage.PdiConection PdiConection { get; set; }
```
Comment (rs22) [inline src/TrainDetectionSystem/Components/TDS/TDSState.cs]: same typo here

File: src/TrainDetectionSystem/TrainDetectionSystem.cs [hunk 3]
```diff
@@ -33,38 +34,47 @@ public class TrainDetectionSystem : BackgroundService
         AsyncDuplexStreamingCall<SciPacket, SciPacket> _currentConnection;
         private Dictionary<string, TvpsOccupancyStatus> _trackSectionStatuses;
 
-        public TrainDetectionSystem(ILogger<TrainDetectionSystem> logger, IConfiguration configuration)
+        private TDSState _tdsState;
+        public TDSState TdsState { get { return _tdsState; } }
```
Comment (rs22) [inline src/TrainDetectionSystem/TrainDetectionSystem.cs]: Nit: you could make this a readonly property (initializable from the constructor only) and save the private variable

`public TDSState TdsState { get; }`

**Review by rs22 — APPROVED** [review]
Thanks, just left a few remarks.

File: src/ProtobufInterfaces/proto/tds.proto (hunk 1 above)
Comment (LeandroParisi) [reply src/ProtobufInterfaces/proto/tds.proto]: solved

File: src/TrainDetectionSystem/Components/TDS/TDSState.cs (hunk 2 above)
Comment (LeandroParisi) [reply src/TrainDetectionSystem/Components/TDS/TDSState.cs]: fixed

File: src/TrainDetectionSystem/TrainDetectionSystem.cs (hunk 3 above)
Comment (LeandroParisi) [reply src/TrainDetectionSystem/TrainDetectionSystem.cs]: solved

**Pull request merged**